pandas/lib.pyx

```diff
@@ -26,9 +26,9 @@ from cpython cimport (PyDict_New, PyDict_GetItem, PyDict_SetItem,
                       PyBytes_GET_SIZE,
                       PyUnicode_GET_SIZE)
 
-try:
+IF PY_MAJOR_VERSION == 2:
```

Comment:
it is more typical for ``C`` code, but converting a ``pyx`` file to ``C ``happens during a cython compiliation, and the only variables available are the ones appearing in cython_compile_time_env AFAICT, which is why I needed to define the variable in setup.py. 

It is not surprising that this is not well known, the only reference I could find on it was a [stack overflow question](http://stackoverflow.com/questions/27273302). I tried all different ways to rewrite this code to keep the ``PyString_GET_SIZE`` on python2 and make it ``PyUnicode_GET_SIZE`` on python3, this was the least intrusive